adam-core/src/main/scala/org/bdgenomics/adam/models/Alphabet.scala

```diff
@@ -0,0 +1,106 @@
+/**
+ * Licensed to Big Data Genomics (BDG) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The BDG licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bdgenomics.adam.models
+
+import scala.util.Try
+
+/**
+ * Created by bryan on 4/17/15.
+ *
+ * An alphabet a symbols and related operations
+ *
+ */
+trait Alphabet {
+
+  /** the symbols in this alphabet */
+  val symbols: Seq[Symbol]
+
+  /**
+   * flag if symbols are case-sensitive.
+   * if true, this alphabet will treat symbols representing upper and lower case symbols as distinct
+   * if false, this alphabet will treat upper or lower case chars as the same for its symbols
+   *
+   */
+  val caseSensitive: Boolean
+
+  /** map of the symbol char to the symbol */
+  lazy val symbolMap: Map[Char, Symbol] =
+    if (caseSensitive)
+      symbols.map(x => x.label -> x).toMap
+    else
+      symbols.flatMap(x => Seq(x.label.toLower -> x.toLower, x.label.toUpper -> x.toUpper)).toMap
+
+  /**
+   *
+   * @param s Each char in this string represents a symbol on the alphabet.
+   *          If the char is not in the alphabet then the unknown char is treated as the complement
+   * @return the reversed complement of the given string.
+   */
+  def reverseComplement(s: String): String = {
+    s.map(x => Try(apply(x)).getOrElse(Symbol(x, x)).complement).reverse
```

Comment:
Ok.  so the Try in this case will catch the KeyNotFoundException and replace it with the Symbol(x,x).  This is to maintain the existing behavior in the SequenceUtils class where an unrecognized character is just kept as is.  The reverseComplementExact will throw the KeyNotFound but I can convert that to an IllegalArgument instead